### DISCUSSION TITLE
cloudflare 奇怪的 ip 地址

### DISCUSSION THREAD

**Discussion by Johnson-LYS** [jeessy2/ddns-go#1250] [General]
在软路由跑了 ddns-go 解析到 cloudflare 一条 a 记录，改记录的 ip 地址会周期性的变成一个 50.114.xx.xx，过几分钟又恢复正常 ip。
检查了 ddns-go 的日志，均没有相关变更，尝试过一下操作：

1. 暂停 ddns-go 服务并删除 a 记录，观察一段时间后并未有新的 a 记录创建
2. 设置 ddns-go 的 ip 获取方式为 api 获取和接口获取，都会出现改问题

想查看 cloudflare 的 dns 修改日志，但是没找到，也没找到 dns 修改时邮件通知的设置。

**Comment by jeessy2** [comment]
多检查，一般是自己的问题。有几个人也提出此问题，都被狠狠打脸

**Reply by Johnson-LYS** [reply]
> 多检查，一般是自己的问题。有几个人也提出此问题，都被狠狠打脸

找到问题原因了，安装了 clash 且通过 api 获取 IP 地址时，https://api.ipify.org 命中了代理规则（用的 shellcrash 默认生成规则），返回了 fake-ip，其它几个获取ip的api都正常

**Comment by Johnson-LYS** [comment]
排查了几天，最后将所有 api token 重置后恢复正常，但是不太确定是不是 api token 泄露，因为排查过程中发现，停止 ddns-go 并删除记录后，不会被创建，但是手动创建记录后，过一会儿记录会被修改。

**Reply by jeessy2** [reply]
估计开了多个ddns软件

**Reply by Johnson-LYS** [reply]
> 估计开了多个ddns软件

我也怀疑这个原因，但是检查了所有机器都没有，而且删除后没有自动创建记录也不太像是跑了其它ddns。留个参考案例吧，类似问题先重置 api token 试试
